src/FffGcodeWriter.cpp

```diff
@@ -1660,48 +1732,80 @@ bool FffGcodeWriter::addSupportInfillToGCode(const SliceDataStorage& storage, La
     EFillMethod support_pattern = infill_extr.getSettingAsFillMethod("support_pattern"); // first layer pattern must be same as other layers
     if (layer_nr <= 0 && (support_pattern == EFillMethod::LINES || support_pattern == EFillMethod::ZIG_ZAG)) { support_pattern = EFillMethod::GRID; }
 
-    int infill_extruder_nr_here = (layer_nr <= 0)? getSettingAsIndex("support_extruder_nr_layer_0") : getSettingAsIndex("support_infill_extruder_nr");
-    const ExtruderTrain& infill_extr_here = *storage.meshgroup->getExtruderTrain(infill_extruder_nr_here);
-
-    const Polygons& support = support_layer.supportAreas;
-
-    std::vector<PolygonsPart> support_islands = support.splitIntoParts();
+    int infill_extruder_nr_here = (layer_nr <= 0) ? getSettingAsIndex("support_extruder_nr_layer_0") : getSettingAsIndex("support_infill_extruder_nr");
 
+    // create a list of outlines and use PathOrderOptimizer to optimize the travel move
+    std::vector<Polygons> infill_outline_list;
```

Comment:
This variable isn't needed. You can just add the outlines in the loop below, where we ad stuff to the `island_order_optimizer`.